amazon-s3-backup/run.sh

```diff
@@ -14,13 +19,29 @@ local_backups_to_keep="$(bashio::config 'local_backups_to_keep' '3')"
 monitor_path="/backup"
 jq_filter=".backups|=sort_by(.date)|.backups|reverse|.[$local_backups_to_keep:]|.[].slug"
 
+
 export AWS_ACCESS_KEY_ID="$(bashio::config 'aws_access_key')"
 export AWS_SECRET_ACCESS_KEY="$(bashio::config 'aws_secret_access_key')"
 export AWS_REGION="$bucket_region"
 
 bashio::log.debug "Using AWS CLI version: '$(aws --version)'"
-bashio::log.debug "Command: 'aws s3 sync $monitor_path s3://$bucket_name/ --no-progress --region $bucket_region --storage-class $storage_class'"
-aws s3 sync $monitor_path s3://"$bucket_name"/ --no-progress --region "$bucket_region" --storage-class "$storage_class"
+
+readarray -t backups < <(bashio::api.supervisor GET /backups/info false .backups | jq -c '.[]')
+for backup in "${backups[@]}"; do
+  bashio::log.debug $(echo "${backup}" | jq -C .)
+
+  slug=$(echo "${backup}" | jq -r .slug)
+  name=$(echo "${backup}" | jq -r .name)
+
+  if ! aws s3 ls "s3://${bucket_name}/${name}.tar" >/dev/null; then
```

Comment:
I think we could/should extract the date information we get from the `/backups/info` response (`date `), and use `year`, `month`, `day`, `hour` as object key prefix. 

For example:
```json
{
  "backups": [
    {
      "slug": "skuwe823",
      "date": "2020-09-30T20:25:34.273Z",
      "name": "Awesome backup with     : some@weird=//characters",
      "type": "partial",
      "size": 44,
      "protected": true,
      "compressed": true,
      "content": {
        "homeassistant": true,
        "addons": ["awesome_addon"],
        "folders": ["ssl", "media"]
      }
    }
  ],
  "days_until_stale": 30
}
```
Together with my previous comment, this would result in:
```
object_key_name="Awesome_backup_with_some-weird-characters"
object_key="$year/$month/$day/$hour/$object_key_name"
# Would result in: '2020/09/30/20/Awesome_backup_with_some-weird-characters.tar'
```

Would also make it easier to manually delete older files via the console or create rules based on the key and we would not need the additional `aws s3 ls` - (although theoretically there could still be an object with the same key, but very very unlikely).